cli/kubectl_kadalu/storage_add.py

```diff
@@ -92,35 +91,52 @@ def storage_add_validation(args):
                    (1 if args.external is not None else 0)
 
     if num_storages == 0:
-        print("Please specify atleast one storage", file=sys.stderr)
+        print("Please specify at least one storage", file=sys.stderr)
         sys.exit(1)
 
-    storage_mismatch = False
-    if (args.type == "Replica1" and num_storages != 1) or (
-            args.type == "Replica3" and num_storages != 3
-    ):
-        storage_mismatch = True
-
-    if args.type == "Replica2" and num_storages != 2:
-        storage_mismatch = True
-
-    if storage_mismatch:
+    # pylint: disable=too-many-boolean-expressions
+    if ((args.type == "Replica1" and num_storages != 1) or
+            (args.type == "Replica2" and num_storages != 2) or
+            (args.type == "Replica3" and num_storages != 3)):
         print("Number of storages not matching for type=%s" % args.type,
               file=sys.stderr)
         sys.exit(1)
 
+    kube_nodes = get_kube_nodes()
+
     for dev in args.device:
         if ":" not in dev:
             print("Invalid storage device details. Please specify device "
                   "details in the format <node>:<device>", file=sys.stderr)
             sys.exit(1)
+        if dev.split(":")[0] not in kube_nodes:
+            print("Node name does not appear to be valid")
+
 
     for path in args.path:
         if ":" not in path:
             print("Invalid storage path details. Please specify path "
                   "details in the format <node>:<path>", file=sys.stderr)
             sys.exit(1)
+        if path.split(":")[0] not in kube_nodes:
+            print("Node name does not appear to be valid")
 
+def get_kube_nodes():
+    """ gets all nodes  """
+    try:
+        #cmd = ["kubectl", "get", "nodes", "--no-headers", "-o", "custom-columns=':metadata.name'"]
+        # above returns <none>
+        cmd = ["kubectl", "get", "nodes"]
+        resp = utils.execute(cmd)
+        print("The following nodes are available")
+        print(resp.stdout)
+        return resp.stdout
```

Comment:
This will have nodename and other details too. Comparing node name will fail while checking.

```
nodes = []
for line in resp.stdout.split("\n"):
    nodename = line.split()[0]
    if nodename != "NAME":
        nodes.append(nodename)
return nodes
```

Reply:
Thanks for the suggestion, `python` is still quite new to me :-O

Reply:
I actually want to use kubcetl, which would be easier, bit somehow it get's me `<empty>` when running in the `cli`

```
cmd = ["kubectl", "get", "nodes", "--no-headers", "-o", "custom-columns=':metadata.name'"]
```